src/Server/Workspace.php

```diff
@@ -200,4 +214,75 @@ public function xdependencies(): array
         }
         return $dependencyReferences;
     }
+
+    /**
+     * Fires when client changes settings in the client
+     *
+     * The default paramter type is Options but it also accepts different types
+     * which will be transformed on demand.
+     *
+     * Currently only the vscode format is supported
+     *
+     * @param mixed|null $settings
+     * @return bool
```

Comment:
You shouldn't add return values just to do assertions when what you really want to test is that certain side effects happen. You should mock classes and then assert that a method is called or not.